controllers/discordactions.js

```diff
@@ -437,14 +439,25 @@ const generateInviteForUser = async (req, res) => {
     const discordInviteResponse = await response.json();
 
     const inviteCode = discordInviteResponse.data.code;
-    const inviteLink = `discord.gg/${inviteCode}`;
+    inviteLink = `discord.gg/${inviteCode}`;
 
-    await discordRolesModel.addInviteToInviteModel({ userId: userIdForInvite, inviteLink });
+    if (dev) {
+      const purpose = req.body.purpose;
+      await discordRolesModel.addInviteToInviteModel({ userId: userIdForInvite, inviteLink, purpose });
 
-    return res.status(201).json({
-      message: "invite generated successfully",
-      inviteLink,
-    });
+      return res.status(201).json({
+        message: "invite generated successfully",
+        inviteLink,
+        purpose,
+      });
+    } else {
+      await discordRolesModel.addInviteToInviteModel({ userId: userIdForInvite, inviteLink });
+
+      return res.status(201).json({
+        message: "invite generated successfully",
+        inviteLink,
+      });
+    }
```

Comment:
No need to use else block here

Reply:
In if block, you are returning. So no need to use else block.

Reply:
done